runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/ActiveWorkState.java

```diff
@@ -168,6 +198,18 @@ synchronized void failWorkForKey(Map<Long, List<FailedTokens>> failedWork) {
     }
   }
 
+  private synchronized void incrementActiveWorkBudget(Work work) {
+    GetWorkBudget currentActiveWorkBudget = activeGetWorkBudget.get();
+    activeGetWorkBudget.set(
+        currentActiveWorkBudget.apply(1, work.getWorkItem().getSerializedSize()));
+  }
+
+  private synchronized void decrementActiveWorkBudget(Work work) {
```

Comment:
should we have some guards against going negative?

Reply:
will never go negative since guards are in `GetWorkBudget`

```
    public final GetWorkBudget build() {
      setItems(Math.max(0, items()));
      setBytes(Math.max(0, bytes()));
      return autoBuild();
    }
```